File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/PacGoat/src/Robot.cpp

#include "Robot.h"

/*
 * All WPILib classes are in the FRC namespace. Either `using namespace frc` 
 * or frc scope (ex. `frc::RobotDrive`) are required to use WPILib functionality. 
 * For headers, use scope instead of `using namespace frc` to avoid global namespace
 * pollution. For source files, either option is viable.
 * See http://wpilib.screenstepslive.com/s/4485/m/13810/l/680719 for more information
 */
using namespace frc;

std::shared_ptr<DriveTrain> Robot::drivetrain;
std::shared_ptr<Pivot> Robot::pivot;
std::shared_ptr<Collector> Robot::collector;
std::shared_ptr<Shooter> Robot::shooter;
std::shared_ptr<Pneumatics> Robot::pneumatics;

std::unique_ptr<OI> Robot::oi;

void Robot::RobotInit() {
	drivetrain.reset(new DriveTrain());
	pivot.reset(new Pivot());
	collector.reset(new Collector());
	shooter.reset(new Shooter());
	pneumatics.reset(new Pneumatics());

	oi.reset(new OI());

	// Show what command your subsystem is running on the SmartDashboard
	SmartDashboard::PutData(drivetrain.get());
	SmartDashboard::PutData(pivot.get());
	SmartDashboard::PutData(collector.get());
	SmartDashboard::PutData(shooter.get());
	SmartDashboard::PutData(pneumatics.get());

	// instantiate the command used for the autonomous period
	autoChooser.AddDefault("Drive and Shoot", driveAndShootAuto.get());
	autoChooser.AddObject("Drive Forward", driveForwardAuto.get());
	SmartDashboard::PutData("Auto Mode", &autoChooser);

	pneumatics->Start(); // Pressurize the pneumatics.
}

void Robot::AutonomousInit() {
	autonomousCommand = (Command *)autoChooser.GetSelected();
	autonomousCommand->Start();
}

void Robot::AutonomousPeriodic() {
	Scheduler::GetInstance()->Run();
	Log();
}

void Robot::TeleopInit() {
	// This makes sure that the autonomous stops running when
	// teleop starts running. If you want the autonomous to
	// continue until interrupted by another command, remove
	// this line or comment it out.
	if (autonomousCommand != nullptr) {
		autonomousCommand->Cancel();
	}
	std::cout << "Starting Teleop" << std::endl;
}

void Robot::TeleopPeriodic() {
	Scheduler::GetInstance()->Run();
	Log();
}

void Robot::TestPeriodic() {
  LiveWindow::GetInstance()->Run();
}

void Robot::DisabledInit() {
	shooter->Unlatch();
}

void Robot::DisabledPeriodic() {
	Log();
}

/**
 * Log interesting values to the SmartDashboard.
 */
void Robot::Log() {
	Robot::pneumatics->WritePressure();
	SmartDashboard::PutNumber("Pivot Pot Value", pivot->GetAngle());
	SmartDashboard::PutNumber("Left Distance", drivetrain->GetLeftEncoder()->GetDistance());
	SmartDashboard::PutNumber("Right Distance", drivetrain->GetRightEncoder()->GetDistance());
}

START_ROBOT_CLASS(Robot)

File: edu.wpi.first.wpilib.plugins.java/resources/templates/examples/PacGoat/src/org/usfirst/frc/team190/pacgoat/Robot.java
package $package;

import edu.wpi.first.wpilibj.IterativeRobot;
import edu.wpi.first.wpilibj.command.Command;
import edu.wpi.first.wpilibj.command.Scheduler;
import edu.wpi.first.wpilibj.livewindow.LiveWindow;
import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

import $package.commands.DriveAndShootAutonomous;
import $package.commands.DriveForward;
import $package.subsystems.Collector;
import $package.subsystems.DriveTrain;
import $package.subsystems.Pivot;
import $package.subsystems.Pneumatics;
import $package.subsystems.Shooter;

/**
 * This is the main class for running the PacGoat code.
 *
 * The VM is configured to automatically run this class, and to call the
 * functions corresponding to each mode, as described in the IterativeRobot
 * documentation. If you change the name of this class or the package after
 * creating this project, you must also update the manifest file in the resource
 * directory.
 */
public class Robot extends IterativeRobot {
	Command autonomousCommand;
	public static OI oi;

	// Initialize the subsystems
	public static DriveTrain drivetrain = new DriveTrain();
	public static Collector collector = new Collector();
	public static Shooter shooter = new Shooter();
	public static Pneumatics pneumatics = new Pneumatics();
	public static Pivot pivot = new Pivot();

	public SendableChooser autoChooser;
	public SendableChooser autonomousDirectionChooser;

	// This function is run when the robot is first started up and should be
	// used for any initialization code.
	@Override
	public void robotInit() {
		SmartDashboard.putData(drivetrain);
		SmartDashboard.putData(collector);
		SmartDashboard.putData(shooter);
		SmartDashboard.putData(pneumatics);
		SmartDashboard.putData(pivot);

		// This MUST be here. If the OI creates Commands (which it very likely
		// will), constructing it during the construction of CommandBase (from
		// which commands extend), subsystems are not guaranteed to be
		// yet. Thus, their requires() statements may grab null pointers. Bad
		// news. Don't move it.
		oi = new OI();

		// instantiate the command used for the autonomous period
		autoChooser = new SendableChooser();
		autoChooser.addDefault("Drive and Shoot", new DriveAndShootAutonomous());
		autoChooser.addObject("Drive Forward", new DriveForward());
		SmartDashboard.putData("Auto Mode", autoChooser);

		pneumatics.start(); // Pressurize the pneumatics.
	}

	@Override
	public void autonomousInit() {
		autonomousCommand = (Command) autoChooser.getSelected();
		autonomousCommand.start();
	}

	// This function is called periodically during autonomous
	@Override
	public void autonomousPeriodic() {
		Scheduler.getInstance().run();
		log();
	}

	@Override
	public void teleopInit() {
		// This makes sure that the autonomous stops running when
		// teleop starts running. If you want the autonomous to
		// continue until interrupted by another command, remove
		// this line or comment it out.
		if (autonomousCommand != null) {
			autonomousCommand.cancel();
		}
	}

	// This function is called periodically during operator control
	@Override
	public void teleopPeriodic() {
		Scheduler.getInstance().run();
		log();
	}

	// This function called periodically during test mode
	@Override
	public void testPeriodic() {
		LiveWindow.run();
	}

	@Override
	public void disabledInit() {
		Robot.shooter.unlatch();
	}

	// This function is called periodically while disabled
	@Override
	public void disabledPeriodic() {
		log();
	}

	/**
	 * Log interesting values to the SmartDashboard.
	 */
	private void log() {
		Robot.pneumatics.writePressure();
		SmartDashboard.putNumber("Pivot Pot Value", Robot.pivot.getAngle());
		SmartDashboard.putNumber("Left Distance", drivetrain.getLeftEncoder().getDistance());
		SmartDashboard.putNumber("Right Distance", drivetrain.getRightEncoder().getDistance());
	}
}
